syft/pkg/cataloger/golang/licenses.go

```diff
@@ -311,3 +312,57 @@ func getModuleRepository(moduleName string, moduleVersion string) (fs.FS, error)
 
 	return billyFSAdapter{fs: f}, nil
 }
+
+type noLicensesFound struct {
+	glob string
+}
+
+func (l noLicensesFound) Error() string {
+	return fmt.Sprintf("unable to find license information matching: %s", l.glob)
+}
+
+var _ error = (*noLicensesFound)(nil)
+
+func toPkgLicenses(goLicenses []goLicense) []pkg.License {
+	var out []pkg.License
+	for _, l := range goLicenses {
+		out = append(out, pkg.License{
+			Value:          l.Value,
+			SPDXExpression: l.SPDXExpression,
+			Type:           l.Type,
+			URLs:           l.URLs,
+			Locations:      toPkgLocations(l.Locations),
```

Comment:
I'm not certain if this is accurate, since these locations might be leaked from the host. Instead, for locations we got remotely should specify the URL and not contain a location https://github.com/anchore/syft/blob/254a562b4e3d5ad6c3f471342ae0107d570b5eb9/syft/pkg/license.go#L30